Cython/Utility/Coroutine.c

```diff
@@ -1501,13 +1501,31 @@ static PyObject *__Pyx_CoroutineAwait_no_new(PyTypeObject *type, PyObject *args,
 }
 #endif
 
+#if PY_VERSION_HEX < 0x03060000 || CYTHON_USE_TYPE_SPECS
+// In earlier versions of Python an object with no __dict__ and not __slots__ is assumed
+// to be pickleable by default. Coroutine-wrappers have significant state so shouldn't be.
+// Therefore provide a default implementation.
```

Comment:
This seems a separate PR as well. I understand that the name change unearthed this issue, but they are not actually otherwise related.